app/routes/synonyms.$synonymId._index.tsx

```diff
@@ -0,0 +1,176 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import type { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node'
+import { redirect } from '@remix-run/node'
+import { Form, Link, useLoaderData } from '@remix-run/react'
+import { Button, ButtonGroup, FormGroup, Icon } from '@trussworks/react-uswds'
+import { useState } from 'react'
+
+import {
+  deleteSynonyms,
+  getSynonymsByUuid,
+  putSynonyms,
+} from './synonyms/synonyms.server'
+import { getFormDataString } from '~/lib/utils'
+
+export async function loader({ params: { synonymId } }: LoaderFunctionArgs) {
+  if (!synonymId) return { synonymId: null, eventIds: null }
```

Comment:
You can use `invariant(synonymId)` to tell TypeScript that it cannot be null or undefined because Remix will not load this route if `synonymId` is an empty string.